aptos-move/framework/aptos-framework/sources/scheduled_transactions/scheduled_txns.move

```diff
@@ -0,0 +1,790 @@
+                BIG_ORDRD_MAP_TGT_ND_SZ / SCHEDULE_MAP_KEY_SIZE,
+                SCHEDULE_MAP_LEAF_DEGREE,
+                true
+            ),
+            txn_tbl: big_ordered_map::new_with_config(
+                (BIG_ORDRD_MAP_TGT_ND_SZ / TXN_ID_SIZE),
+                (BIG_ORDRD_MAP_TGT_ND_SZ / (TXN_ID_SIZE + AVG_SCHED_TXN_SIZE)),
+                true
+            )
+        };
+        move_to(framework, queue);
+
+        // Parallelizable data structure used to track executed txn_ids.
+        move_to(
+            framework,
+            ToRemoveTbl {
+                remove_tbl: table::new<u16, vector<TransactionId>>()
+            }
+        );
+    }
+
+    /// Stop, remove and refund all scheduled txns; can be called only by the framework
+    public fun shutdown(
+        framework: &signer
+    ) acquires ScheduleQueue, ToRemoveTbl, GasFeeDepositStoreSignerCap {
+        system_addresses::assert_aptos_framework(framework);
+
+        // Make a list of txns to cancel and refund
+        let txns_to_cancel = vector::empty<ScheduledTransactionWithId>();
+        let queue = borrow_global<ScheduleQueue>(signer::address_of(framework));
+        let iter = queue.txn_tbl.new_begin_iter();
+        while (!iter.iter_is_end(&queue.txn_tbl)) {
+            let txn_id = *iter.iter_borrow_key();
+            let txn = iter.iter_borrow(&queue.txn_tbl);
+            let scheduled_txn_with_id = ScheduledTransactionWithId { txn: *txn, txn_id };
+            txns_to_cancel.push_back(scheduled_txn_with_id);
+            iter = iter.iter_next(&queue.txn_tbl);
+        };
+
+        // Cancel all transactions
+        while (!txns_to_cancel.is_empty()) {
+            let txn_with_id = txns_to_cancel.pop_back();
+            // Create a new signer from the stored handle
+            let schedule_txn_signer =
+                permissioned_signer::signer_from_storable_permissioned_handle(
+                    &txn_with_id.txn.sender_handle
+                );
+            cancel(&schedule_txn_signer, txn_with_id.txn_id.hash);
+        };
+
+        // Remove and destroy resources
+        let ScheduleQueue { schedule_map, txn_tbl } =
+            move_from<ScheduleQueue>(signer::address_of(framework));
+        schedule_map.destroy(|_| {});
+        txn_tbl.destroy(|_| {});
+
+        // Clean up ToRemoveTbl; we can only empty the table but not drop it!
+        let ToRemoveTbl { remove_tbl } =
+            borrow_global_mut<ToRemoveTbl>(signer::address_of(framework));
+        let i = 0;
+        while (i < TO_REMOVE_PARALLELISM) {
+            if (remove_tbl.contains((i as u16))) {
+                remove_tbl.remove((i as u16));
+            };
+            i = i + 1;
+        };
+    }
+
+    /// todo: Do we need a function to pause ???
+
+    /// Insert a scheduled transaction into the queue. Txn_id is returned to user, which can be used to cancel the txn.
+    public fun insert(
+        sender: &signer, txn: ScheduledTransaction
```

Comment:
since the signer handle is already in the ScheduledTransaction, do we still need the sender? for gas deposit we can accept a FungibleAsset directly?